### ISSUE TITLE
Resource leaks in apps.c

### ISSUE THREAD

**Issue by ryancaicse** [openssl/openssl#17330]
Should **fp** be closed before **Line 3045**?
https://github.com/openssl/openssl/blob/79b2a2f2eedb9d6b24a3f6748332328cf54568fb/apps/lib/apps.c#L3036-L3059

**Comment by t8m** [comment]
No, why do you think so? If b is non-null the fp is owned by the BIO.

**Comment by ryancaicse** [comment]
@t8m Thank you.

I misunderstand the code. **fp** should only be closed when going the `err`.

https://github.com/openssl/openssl/blob/79b2a2f2eedb9d6b24a3f6748332328cf54568fb/apps/lib/apps.c#L3043-L3058
